src/plugins/upf/upf_pfcp.c

```diff
@@ -733,9 +734,14 @@ sx_make_pending_pdr (upf_session_t * sx)
       vec_foreach_index (i, active->pdr)
       {
 	upf_pdr_t *pdr = vec_elt_at_index (pending->pdr, i);
+        acl_rule_t *acl, *acl_rule;
 
-	pdr->pdi.adr.db_id = upf_adf_get_adr_db (pdr->pdi.adr.application_id);
+	pdr->pdi.adr.db_id = upf_adf_get_adr_db (pdr->pdi.adr.application_id, &acl);
 	pdr->pdi.acl = vec_dup (vec_elt (active->pdr, i).pdi.acl);
+        vec_foreach (acl_rule, acl)
+          {
+            vec_add1(pdr->pdi.acl, *acl_rule);
+          }
```

Comment:
This looks wrong. If you reload the ACLs like this, then you also need to keep the PFCP specified SDFs somewhere.

Reply:
The vector is not reset there. The rules from the app are added to it, so both PFCP-specified SDFs and ones from the app apply.

Reply:
yes, but in handle_{create,update}_pdr you already adding the ACLs from the application to the PDI. Doing here again will duplicate the application ACL.

Reply:
Yes, you're right. Removed this part